docs/reference/pyboard.py.rst

```diff
@@ -48,6 +52,10 @@ Running ``pyboard.py --help`` gives the following output:
                             available
       --follow              follow the output after running the scripts
                             [default if no scripts given]
+      --no-soft-reset       Prevent performing a soft reset when connecting to MCU
+      --hard-reset          pulse the MCU reset pin to hard-reset the MCU (if your
+                            serial connection wires RTS to reset pin properly)
```

Comment:
This scheme of wiring RTS to a reset pin is _only_ used on a handful of esp32 boards, not any newer esps with built in USB, nor on any other ports.

"properly" doesn't make sense in this help message here.

Also keep in mind that pyboard.py is generally not expected to be used directly by end users any more, mpremote is the standard / promoted tool now.

Reply:
> not any newer esps with built in USB, nor on any other ports.

On that - do you happen to know *how* any of those other boards reset the ESP32 and assert GPI0 for flashing, *without* using RST/DSR? (assuming any of them do, and it's not necessary to physically push buttons on them for it).  If there's some other trick to hard-reset modern inbuilt-USB boards, it would be worth including that feature (and porting all this to mpremote too... I've not tried it, but I expect mpremote has the same serial issue, ... time passes... yes: /mpremote/transport_serial.py is a copy/paste of pyboard.py )

Reply:
Most / many other mcu boards just have a separate jumper / button on the board to jump into upgrade mode - every MCU is generally different here.
Some other MCU's with built in USB have either internal mechanisms to run a bootloader.

It's not a standard feature by any account to have a hardware handled reset function over usb, 
The esp board (arguably abusing) the flow control pins to do this is the first time I've ever seen this done; and it often causes conflicts where the serial port doesn't work properly on applications that try to enable flow control be default.

Indeed, having the support for this esp RTS/CTS pin control in pyboard/mpremote has broken support for some stm32 boards which enable flow control for reliable serial transfers, I'd personally prefer it was removed entirely, or at least not used by default!

Reply:
> This scheme of wiring RTS to a reset pin is only used on a handful of esp32 boards, not any newer esps with built in USB, nor on any other ports.

Not even all "traditional" ESP boards are the same. Same vendors save a few cent and do not add the two transistor logic to RTS/DTR, but connect Reset & GPIO0 directly to RTS and DTR. That causes problems. Likewise, not all boards with built-in USB switch to boot loader mode when the RTS/DTR toggling is applied. ESP32C3 do, ESP32S3/ESP32S2  do not.

Reply:
> broken support for some stm32 boards

I own a Roland MDX-20 CNC machine with hardware flow control, so I'm *extremely* aware of how badly supported this idea has been over the decades (and indeed - a google of stm32 hardware flow control leads with a pile of people having problems...) but I digress.

I've got a huge selection of ESP32 and ESP8266 boards which I test on (and rPI which I own, but haven't tested with yet): I think I should buy a selection of other kinds so I can better check that my "fix" (which is in fact just making linux behave the way windows already does now with serial port RTS pins) has no negative impacts on other architectures/boards... can you point me at the one(s) I need to get (preferably aliexpress if they're there) to add to my collection ?